src/sentry/integrations/vercel/client.py

```diff
@@ -82,28 +82,32 @@ def create_deploy_webhook(self):
         return response
 
     def get_env_vars(self, vercel_project_id):
-        return self.paginate(self.GET_ENV_VAR_URL % vercel_project_id, "envs")
+        return self.get(self.GET_ENV_VAR_URL % vercel_project_id)
 
     def create_secret(self, vercel_project_id, name, value):
         data = {"name": name, "value": value}
         response = self.post(self.SECRETS_URL, data=data)["uid"]
         return response
 
     def create_env_variable(self, vercel_project_id, key, value):
-        data = {"key": key, "value": value, "target": "production"}
-        response = self.post(self.ENV_VAR_URL % vercel_project_id, data=data)
-        return response
-
-    def update_env_variable(self, vercel_project_id, key, value):
+        data = {"key": key, "value": value, "target": ["production"], "type": "secret"}
         try:
-            self.delete(
-                self.DELETE_ENV_VAR_URL % (vercel_project_id, key),
-                allow_text=True,
-                params={"target": "production"},
-            )
+            return self.post(self.ENV_VAR_URL % vercel_project_id, data=data)
         except ApiError as e:
-            # we can ignore 404 errors here since we are just trying to delete
-            if e.code != 404:
-                raise
-
-        return self.create_env_variable(vercel_project_id, key, value)
+            if e.json:
+                if e.json["error"]:
+                    if e.json["error"]["code"] == "ENV_ALREADY_EXISTS":
+                        return self.update_env_variable(vercel_project_id, key, value, data)
+            raise
+
+    def update_env_variable(self, vercel_project_id, key, value, data):
+
+        env_var_id = [
+            env_var["id"]
+            for env_var in self.get(self.GET_ENV_VAR_URL % vercel_project_id)["envs"]
+            if env_var["key"] == key
+        ]
+        if env_var_id:
```

Comment:
@ceorourke we should throw some sort of error if `env_var_id` is falsy because that means we failed to actually update the project correctly.

Reply:
@scefali you think I should log this failure too? I think it'd be good to know.

Reply:
@ceorourke @scefali yeah I think logging could be helpful here. Im worried that for the recent bug, it seemed like we weren't able to find the env var we were looking for (since we didn't try to delete and recreate) so if we didn't find it, that means we also won't get the `env_var_id` in order to update it.

Reply:
@ceorourke I would prefer an error like this to go to Sentry rather than logging. Generally, we reserve logging for errors that are bad but are expected to happen. In this case, I would always expect `update_env_variable` to find the `env_var_id` that matches if we are getting any sort of response back from Vercel. Does that make sense?